nixos/modules/services/networking/nbd.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nbd;
+  serverConfig = pkgs.writeText "nbd-server-config" (''
+    [generic]
+    user = root
+    group = root
+    allowlist = false
+    listenaddr = ${cfg.server.listenAddress}
+    port = ${toString cfg.server.listenPort}
+  '' + concatStringsSep "\n"
+    (mapAttrsToList
+      (name: { device, authSubnet }:
+        ''
+          [${name}]
+          exportname = ${device}
+          authfile = ${pkgs.writeText "authfile" ''
+            ${authSubnet}
+          ''}
+          flush = true
+          fua = true
+        '')
+      cfg.server.exports));
+in
+{
+  imports = [ ];
+
+  options = {
+    services.nbd = {
+      server = {
+        enable = mkEnableOption "the Network Block Device (nbd) server.";
+
+        listenPort = mkOption {
+          type = types.port;
+          default = 10809;
+          description = "Port to listen on. The port is NOT automatically opened in the firewall.";
+        };
+
+        exports = mkOption {
+          description = "Block devices to make available over the network.";
+          default = { };
+          type = with types; attrsOf
+            (submodule {
+              options = {
+                device = mkOption {
+                  type = types.str;
+                  description = "Block device to export.";
+                  example = "/dev/sdb1";
+                };
+
+                authSubnet = mkOption {
+                  type = types.str;
+                  example = "10.10.0.0/24";
+                  description = "Subnet that is authorized to connect for this device.";
+                };
+              };
+            });
+        };
+
+        listenAddress = mkOption {
```

Comment:
this should probably allow `null` to allow default nbd-server behaviour of listening on everything. might even be a good idea to make that the default.

Reply:
Done and it's now the default.

I did something similar to the `authSubnet` option: it's now called `allowAddresses`, it's a list, and it defaults to `null`, which means anyone is allowed to connect to the export.

There are warnings in the docs about the default values allowing everything.